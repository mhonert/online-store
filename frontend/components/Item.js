import React from 'react';
import PropTypes from 'prop-types';
import ItemStyles from "./styles/ItemStyles";
import Title from "./styles/Title";
import Link from 'next/link';
import PriceTag from "./styles/PriceTag";
import formatMoney from "../lib/formatMoney";

const Item = ({item}) => {
  return (
    <ItemStyles>
      {item.image && <img src={item.image} alt={item.title} />}
      <Title>
        <Link href={{
          pathname: '/item',
          query: { id: item.id }
        }}>
          <a>{item.title}</a>
        </Link>
      </Title>
      <PriceTag>{formatMoney(item.price)}</PriceTag>
      <p>{item.description}</p>

      <div className="buttonList">
        <Link href={{
          pathname: "update",
          query: { id: item.id }
        }}><a>Edit ✏️</a></Link>
        <Link href={{
          pathname: "add",
          query: { id: item.id }
        }}><a>Add to Cart</a></Link>
        <Link href={{
          pathname: "delete",
          query: { id: item.id }
        }}><a>Delete</a></Link>
      </div>
    </ItemStyles>
  );
};

Item.propTypes = {
  item: PropTypes.object.isRequired,
};

export default Item;
