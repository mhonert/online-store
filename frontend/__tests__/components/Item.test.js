import Item from '../../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
  id: '4923423AKDJA',
  title: 'An Item',
  price: 500,
  description: 'This is a test item',
  image: 'item.jpg',
  largeImage: 'largeImage.jpg',
};

describe('<Item/>', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<Item item={fakeItem} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders the pricetag and title', () => {
    const wrapper = shallow(<Item item={fakeItem} />);
    const PriceTag = wrapper.find('PriceTag');

    expect(PriceTag.children().text()).toBe('$5');
    expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
  });

  it('renders the buttons', () => {
    const wrapper = shallow(<Item item={fakeItem} />);
    const buttonList = wrapper.find('.buttonList');

    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find('Link').exists()).toBe(true);
    expect(buttonList.find('AddToCart').exists()).toBe(true);
    expect(buttonList.find('DeleteItem').exists()).toBe(true);
  });
});