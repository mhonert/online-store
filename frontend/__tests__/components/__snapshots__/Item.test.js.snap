// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="An Item"
    src="item.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "4923423AKDJA",
          },
        }
      }
    >
      <a>
        An Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $5
  </PriceTag>
  <p>
    This is a test item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "4923423AKDJA",
          },
        }
      }
    >
      <a>
        Edit ðŸ–Š
      </a>
    </Link>
    <AddToCart
      id="4923423AKDJA"
    />
    <DeleteItem
      id="4923423AKDJA"
    >
      Delete ðŸ—‘
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
